File: backend/app/models/story.py
from pydantic import BaseModel

from app.models.metadata import StoryMetadata
from app.models.provenance import Provenance
from app.models.scene import Scene


class Story(BaseModel):
    story_id: str
    title: str
    universe: str
    character_name: str
    role: str
    mood: str
    story: str
    scenes: list[Scene]
    metadata: StoryMetadata
    provenance: Provenance | None = None


class StoryCard(BaseModel):
    story_id: str
    title: str
    thumbnail: str

File: frontend/types/story.ts
import type { Scene } from "./scene";
import type { StoryMetadata } from "./metadata";

export interface Story {
  story_id: string;
  title: string;
  universe: string;
  character_name: string;
  role: string;
  mood: string;
  story: string;
  scenes: Scene[];
  metadata: StoryMetadata;
}

export interface StoryCard {
  story_id: string;
  title: string;
  thumbnail: string;
}

export interface GenerationRequest {
  universe: string;
  character_name: string;
  role: string;
  mood: string;
  prompt: string;
}

export interface ContinueStoryRequest {
  prompt: string;
}

export interface RegenerateSceneRequest {
  scene_number: number;
}
